code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -360,13 +360,14 @@
 	src.verbs -= silicon_subsystems
 
 /mob/living/silicon/robot/drone/update_canmove(delay_action_updates = 0)
+	. = ..()
+	density = emagged ? TRUE : FALSE //this is reset every canmove update otherwise
+
+/mob/living/silicon/robot/drone/Life(seconds, times_fired)
```

Comment:
While this does work, I'm going to suggest a different solution.

In `/mob/living/silicon/robot/drone/emag_act()`, somewhere after the drone has actually been emagged, you can do:
```dm
addtimer(CALLBACK(src, .proc/shut_down, TRUE), EMAG_TIMER)
```
Then you can remove the drone `Life` proc and also, you can remove the `emagged_time` var from drones, since it won't be needed anymore.

Reply:
added your solution